core/src/main/java/org/infinispan/filter/CompositeKeyValueFilter.java

```diff
@@ -29,6 +26,17 @@ public CompositeKeyValueFilter(KeyValueFilter<? super K, ? super V>... filters)
       this.filters = filters;
```

Comment:
wouldn't be simpler to have a `MarshallableCollection` as a field?

Reply:
Probably. This class is never used internally. Is there any reason to keep it around @wburns or can we deprecate in 15.2 and remove in 16.0?

Reply:
It was used at one point and is technically public. We could try deprecating if you want, but I doubt we will be able to remove it anytime soon and it shouldn't cause any maintenance head ache as it is pretty trivial.

Reply:
I've updated the class to use `MarshallableArray`. We can't use `MarshallableCollection` without copying as the constructor takes in varargs.